core/src/main/java/org/testcontainers/utility/ResourceReaper.java

```diff
@@ -70,118 +55,129 @@ private ResourceReaper() {
+                    .withBinds(Arrays.asList(
+                        new Bind(
+                            DockerClientFactory.instance().getRemoteDockerUnixSocketPath(),
+                            new Volume("/var/run/docker.sock")
+                        )
+                    ))
+                    .withPrivileged(TestcontainersConfiguration.getInstance().isRyukPrivileged())
+            )
+            .withName("testcontainers-ryuk-" + DockerClientFactory.SESSION_ID)
+            .withLabels(Collections.singletonMap(DockerClientFactory.TESTCONTAINERS_LABEL, "true"))
+            .exec()
+            .getId();
 
         StringBuilder ryukLog = new StringBuilder();
 
-        ResultCallback.Adapter<Frame> logCallback = client.logContainerCmd(ryukContainerId)
-            .withSince(0)
+        CountDownLatch started = new CountDownLatch(1);
+
+        log.debug("Attaching to the Ryuk container with id: '{}'", ryukContainerId);
+        ResultCallback.Adapter<Frame> attachment = client.attachContainerCmd(ryukContainerId)
             .withFollowStream(true)
-            .withStdOut(true)
+            .withLogs(true)
             .withStdErr(true)
+            .withStdOut(true)
             .exec(new ResultCallback.Adapter<Frame>() {
                 @Override
                 public void onNext(Frame frame) {
-                    ryukLog.append(new String(frame.getPayload(), StandardCharsets.UTF_8));
+                    String payload = new String(frame.getPayload(), StandardCharsets.UTF_8);
+                    // Poor Man's log-based wait strategy :)
+                    if (frame.getStreamType() == StreamType.STDERR && payload.contains("Started!")) {
+                        started.countDown();
+                    }
+                    ryukLog.append(payload);
                 }
             });
-
-        InspectContainerResponse inspectedContainer = client.inspectContainerCmd(ryukContainerId).exec();
-
-        Integer ryukPort = inspectedContainer.getNetworkSettings().getPorts().getBindings().values().stream()
-                .flatMap(Stream::of)
-                .findFirst()
-                .map(Ports.Binding::getHostPortSpec)
-                .map(Integer::parseInt)
-                .get();
-
-        CountDownLatch ryukScheduledLatch = new CountDownLatch(1);
-
-        synchronized (DEATH_NOTE) {
-            DEATH_NOTE.add(
-                    DockerClientFactory.DEFAULT_LABELS.entrySet().stream()
-                            .<Map.Entry<String, String>>map(it -> new SimpleEntry<>("label", it.getKey() + "=" + it.getValue()))
-                            .collect(Collectors.toList())
-            );
+        if (!attachment.awaitStarted(TestcontainersConfiguration.getInstance().getRyukTimeout(), TimeUnit.SECONDS)) {
+            log.error("Timed out waiting for Ryuk container's attach. Container id: '{}'", ryukContainerId);
+            throw new IllegalStateException("Failed to attack too Ryuk");
```

Comment:
Tiny typos!
```suggestion
            throw new IllegalStateException("Failed to attach to Ryuk");
```

Reply:
omg :D thanks!

Reply:
what a combo of Mac's keyboard and my late evening brain 😂